lib/puppet_x/puppet-community/http_validator.rb

```diff
@@ -0,0 +1,49 @@
+require 'puppet/network/http_pool'
+
+module PuppetX
+  module PuppetCommunity
+    class HttpValidator
+      attr_reader :http_server
+      attr_reader :http_port
+      attr_reader :use_ssl
+      attr_reader :test_path
+      attr_reader :test_headers
+
+      def initialize(http_resource_name, http_server, http_port, use_ssl, test_path)
+       begin
+          uri = URI(http_resource_name)
```

Comment:
At least on Ruby 1.9, URI("string") will always return successful, even if the URI wasn't parsed.  You need to do something like this instead:

    if http_resource_name =~ /\A#{URI::regexp}\z/